### PR TITLE
Migration: Remove LegacyForms from dashboard folder permissions

### LINKED ISSUE #25841: Chore: Remove LegacyForms from Grafana core
A followup on #22862

After the new forms were implemented we were left with a few places where we still used the LegacyForms. These need to go.

[See the current amount of LegacyForms in use](https://play.grafana.org/d/VEy5UXjmk/grafana-ci-code-stats?orgId=1)

To find where they are used, simply search `LegacyForms` in the codebase. Some of these might need some further UX work though. To handle possible API changes, have a look at Forms in [@grafana/ui](https://developers.grafana.com/ui/canary/index.html?path=/story/*)

- [x] Dashboard folder permissions #28564 
- [x] Dashboard feature #28707

There are still some (internal) components in @grafana/ui that does not import LegacyForms, but rather import the sepaate components as a relative path. These are the components:

- [ ] ClipboardButton
- [ ] TagsInput (Where is this even used?)
- [ ] ColorPicker (The input field and Y-axis Switch)

These need more extensive UX/UI work
- [ ] DataSourceHttpSettings
- [ ] CustomHeadersSettings

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR removes the usage of LegacyForms from Dashboard Folder Permissions view. It also restricts the line-height of the grafana-ui `<Select>` component option descriptions as wrapping components / css classes were affecting it.

Screenshot 2020-10-27 at 13 09 35

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
Looks good @jackw!

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ export const getSelectStyles = stylesFactory((theme: GrafanaTheme) => {
       font-size: ${theme.typography.size.sm};
       color: ${theme.colors.textWeak};
       white-space: normal;
+      line-height: ${theme.typography.lineHeight.md};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Select/getSelectStyles.ts:46]: is this not the default body line height?

File: public/app/core/components/PermissionList/AddPermission.tsx [hunk 2]
```diff
@@ -88,56 +86,52 @@ class AddPermissions extends Component<Props, NewDashboardAclItem> {
...
+                  <UserPicker onSelected={this.onUserSelected} className={pickerClassName} />
+                </InlineField>
+              ) : null}
+
+              {newItem.type === AclTarget.Team ? (
+                <InlineField>
+                  <TeamPicker onSelected={this.onTeamSelected} className={pickerClassName} />
+                </InlineField>
+              ) : null}
+
```
Comment (torkelo) [inline public/app/core/components/PermissionList/AddPermission.tsx:119]: A label here that said "Can"  would be nice, to make it similar to table and not just be 3 selects in a row

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts (hunk 1 above)
Comment (jackw) [reply packages/grafana-ui/src/components/Select/getSelectStyles.ts:46]: Yep, that's correct. I guess it's more of a reset. My reasoning for adding it was due to `.filter-table` class used on the permissions list declares [a line-height](https://github.com/grafana/grafana/blob/master/public/sass/components/_filter-table.scss#L33). I considered reworking the table styles but thought it better if the dropdown descriptions looked consistent wherever they're used.

I spoke with @jessover9000 before adding it and we came to the conclusion that of the lineHeight theme values we have right now this one looks the best.

File: public/app/core/components/PermissionList/DisabledPermissionListItem.tsx [hunk 3]
```diff
@@ -28,7 +27,7 @@ export default class DisabledPermissionListItem extends Component<Props, any> {
             <Select
               options={dashboardPermissionLevels}
               onChange={() => {}}
-              isDisabled={true}
+              disabled={true}
               className="gf-form-select-box__control--menu-right"
```
Comment (Clarity-89) [inline public/app/core/components/PermissionList/DisabledPermissionListItem.tsx:31]: Could we maybe also get rid of this class here? Doesn't seems like it has any effect.

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Select/getSelectStyles.ts:46]: ok, yea that filter table line-height is probably not the best solution, but I see why this is needed now

File: public/app/core/components/PermissionList/AddPermission.tsx [hunk 4]
```diff
@@ -88,59 +88,57 @@ class AddPermissions extends Component<Props, NewDashboardAclItem> {
...
+              </Button>
+            </HorizontalGroup>
+          )}
+        </Form>
       </div>
     );
   }
 }
 
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
```
Comment (dprokop) [inline public/app/core/components/PermissionList/AddPermission.tsx:137]: FYI: you don't need to use styles factory. You can define the style getter and use [`useStyles` hook](https://github.com/grafana/grafana/blob/master/contribute/style-guides/styling.md#styling-with-theme) that will provide the theme automatically and guarantee memoization

**Review by dprokop — APPROVED** [review]

File: public/app/core/components/PermissionList/AddPermission.tsx (hunk 4 above)
Comment (jackw) [reply public/app/core/components/PermissionList/AddPermission.tsx:137]: Maybe I'm misunderstanding but to the best of my knowledge hooks can't be used in class components. Did I miss something? 🤔

Comment (dprokop) [reply public/app/core/components/PermissionList/AddPermission.tsx:137]: Nope, it we me missing that this component is a class one. Ignore my comment :)

**Pull request merged**